runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java

```diff
@@ -205,18 +273,93 @@ public void onGameObjectChanged(GameObjectChanged event)
 	@Subscribe
 	public void onGameObjectDespawned(GameObjectDespawned event)
 	{
+		if (!isInMLM())
+		{
+			return;
+		}
+
 		GameObject gameObject = event.getGameObject();
 		rocks.remove(gameObject);
 	}
 
 	@Subscribe
 	public void onGameStateChanged(GameStateChanged event)
 	{
+		if (!isInMLM())
+		{
+			return;
+		}
+
 		if (event.getGameState() == GameState.LOADING)
 		{
 			// on region changes the tiles get set to null
 			veins.clear();
 			rocks.clear();
 		}
 	}
+
+	public Integer getDepositsLeft()
+	{
+		if (MAX_SACK_SIZE == 0) // check if MAX_SACK_SIZE has been initialized
+		{
+			return null;
+		}
+
+		double depositsLeft = 0;
+		Integer nonPayDirtItems = 0;
+
+		Item[] result = queryRunner.runQuery(new InventoryItemQuery(InventoryID.INVENTORY));
+
+		assert result != null;
+
+		int MAX_INVENTORY_SIZE = result.length;
+
+		for (Item item : result)
+		{
+			// Assume that MLM ores are being banked and exclude them from the check,
+			// so the user doesn't see the Overlay switch between deposits left and N/A.
+			//
+			// Count other items at nonPayDirtItems so depositsLeft is calculated accordingly.
+			if (item.getId() != ItemID.PAYDIRT && item.getId() != -1 && !MLM_ORE_TYPES.contains(item.getId()))
+			{
+				nonPayDirtItems += 1;
+			}
+		}
+
+		double inventorySpace = MAX_INVENTORY_SIZE - nonPayDirtItems;
+		double sackSizeRemaining = MAX_SACK_SIZE - CUR_SACK_SIZE;
+
+		if (inventorySpace > 0 && sackSizeRemaining > 0)
+		{
+			depositsLeft = Math.ceil(sackSizeRemaining / inventorySpace);
+
+		}
+		else if (inventorySpace == 0)
+		{
+			return null;
+		}
+
+		return (int) depositsLeft;
+	}
+
+	public boolean isInMLM()
+	{
+		WorldPoint localWorld;
+
+		try
+		{
+			localWorld = client.getLocalPlayer().getWorldLocation();
+		}
+		catch (NullPointerException ex)
+		{
+			return false;
+		}
+
+		int regionX = localWorld.getX() >> 6;
+		int regionY = localWorld.getY() >> 6;
+		int regionID = regionX << 8 | regionY;
+
+		return MOTHERLODE_MAP_REGIONS.contains(regionID);
```

Comment:
There is `client.getMapRegions()`. Maybe it is worth putting a on region change event somewhere. We do have the gamestate change event which will go between region and logged in when the region changes.

Reply:
Now that I think about this, just checking this when the gamestate changes to LOGGED_IN would work fine.